lib/src/rules/pub/secure_pubspec_urls.dart

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/src/lint/pub.dart'; // ignore: implementation_imports
+
+import '../../analyzer.dart';
+
+const _desc = r'Use secure urls in `pubspec.yaml`.';
+
+const _details = r'''
+**DO** Use secure urls in `pubspec.yaml`.
+
+Use `https` instead of `http` an `git:`.
+
+**GOOD:**
+```yaml
+repository: 'https://github.com/dart-lang/example'
+```
+
+**BAD:**
+```dart
+repository: 'http://github.com/dart-lang/example'
+''';
+
+class PubspecNoHttpUrls extends LintRule {
```

Comment:
We generally name the class to be the same as the lint name. It's not critical, and it could be done in a later PR, but I'm mentioning it in case you intended to do so and just forgot.

Reply:
Yes, please.  This makes things much easier to keep track of.